src/macsniff.cpp

```diff
@@ -0,0 +1,145 @@
+#include <BLEDevice.h>
+#include <BLEUtils.h>
+#include <BLEScan.h>
+#include <BLEAdvertisedDevice.h>
+#endif
+
+#ifdef VENDORFILTER
+  #include <array>
+  #include <algorithm>
+  #include "vendor_array.h"
+#endif
+
+// Local logging tag
+static const char *TAG = "macsniff";
+
+static wifi_country_t wifi_country = {.cc=WIFI_MY_COUNTRY, .schan=WIFI_CHANNEL_MIN, .nchan=WIFI_CHANNEL_MAX, .policy=WIFI_COUNTRY_POLICY_MANUAL};
+
+uint16_t currentScanDevice = 0;
+
+bool mac_add(uint8_t *paddr, int8_t rssi, bool sniff_type) {
+
+    char counter [11]; // uint32_t -> 4 byte -> 10 decimals + '0' terminator -> 11 chars
+    char macbuf [21]; // uint64_t -> 8 byte -> 20 decimals + '0' terminator -> 21 chars
+    char typebuff[8] ;
+    bool added = false;
+    uint32_t hashedmac, vendor2int;
+    uint64_t addr2int;
+    std::pair<std::set<uint32_t>::iterator, bool> newmac;
+
+    addr2int = ( (uint64_t)paddr[0] ) | ( (uint64_t)paddr[1] << 8 ) | ( (uint64_t)paddr[2] << 16 ) | \
+    ( (uint64_t)paddr[3] << 24 ) | ( (uint64_t)paddr[4] << 32 ) | ( (uint64_t)paddr[5] << 40 );
+
+    #ifdef VENDORFILTER
+    vendor2int = ( (uint32_t)paddr[2] ) | ( (uint32_t)paddr[1] << 8 ) | ( (uint32_t)paddr[0] << 16 );
+    // No vendor filter for BLE
+    if ( (sniff_type==MAC_SNIFF_BLE) || std::find(vendors.begin(), vendors.end(), vendor2int) != vendors.end() ) {
+    #endif
+
+        //if (!(addr2int & WIFI_MAC_FILTER_MASK)) { // filter local and group MACs   
+
+        // salt and hash MAC, and if new unique one, store hash in container and increment counter on display
+        addr2int <<= 8;
+        addr2int += salt; // append salt value to MAC before hashing it
+        itoa(addr2int, macbuf, 10); // convert 64 bit MAC to base 10 decimal string
+        hashedmac = rokkit(macbuf, 10); // hash MAC for privacy, use 10 chars to fit in uint32_t container
+
+        if (sniff_type == MAC_SNIFF_WIFI ) {
+            newmac = wifis.insert(hashedmac); // store hashed MAC if new unique
+            strcpy(typebuff, "WiFi");
+        } else if (sniff_type == MAC_SNIFF_BLE ) {
+            newmac = bles.insert(hashedmac); // store hashed MAC if new unique
+            strcpy(typebuff, "BLE ");
+        }
+
+        if (newmac.second) { // first time seen this WIFI/BLE MAC
+            // Insert to glbal counter
+            macs.insert(hashedmac);
+            added = true;
+            itoa(macs.size(), counter, 10); // base 10 decimal counter value
+            u8x8.draw2x2String(0, 0, counter);
+            ESP_LOGI(TAG, "%s RSSI %04d -> Hash %010u -> #%04i", typebuff, rssi, hashedmac, macs.size());
+        } else {
+            ESP_LOGI(TAG, "%s RSSI %04d -> already seen", typebuff, rssi);
+        }
+
+    #ifdef VENDORFILTER
+    } else {
+        // Very noisy
+        //ESP_LOGI(TAG, "Filtered MAC %02X:%02X:%02X:%02X:%02X:%02X", paddr[0],paddr[1],paddr[2],paddr[3],paddr[5],paddr[5]);
+    }
+    #endif
+
+    // True if MAC (WiFi/BLE was new)
+    return added;
```

Comment:
return `newmac.second` should give same result, so we don't need extra bool variable here.

Reply:
Correct ;-)